pkg/sql/ir_generator.go

```diff
@@ -440,13 +441,23 @@ func parseExpression(e interface{}) (interface{}, error) {
 					} else {
 						list = append(list, intVal)
 					}
-				} else {
-					value, err := parseExpression(&expr.Sexp)
-					if err != nil {
-						return nil, err
+					continue
+				}
+
+				// parse negative integer
+				if len(expr.Sexp) == 2 && (*expr.Sexp[0]).Value == negative {
```

Comment:
Can add a comment to reference extended_syntax_parser.y:368 for further reading.

Reply:
Same above.